### PR TITLE
feat(events): include `profile` on `signIn` events

### PR DESCRIPTION
**Pull request by t3dotgg**
Redo of https://github.com/nextauthjs/next-auth/pull/2354 rebased on `next` branch

_below is copy-pasted from last PR_

## What Changed
`profile` added to `event.signIn` message

## Reasoning 💡

I use the profile information to fire updates on the user object after any successful sign-in. Before this change, I had to stuff handlers to sync `profile.name` and `platform` against the `User` model in 3 separate places. This change allows me to have a complete "source of truth" within the `signIn` event :)

## Checklist 🧢

To check an item, place an `x` in the box like so: `- [x] Documentation`. -->

- [x] Documentation (likely needed, will add if y'all like the changes)
- [ ] Tests (not sure if needed for this)
- [x] Ready to be merged

I would like some thoughts from the team on this solution before cleanup + merge. I am happy to make any docs changes and add tests if this is a solution y'all are happy with :)

## Affected issues 🎟

Didn't file one, went straight to source :)

### CODE REVIEW

**Comment by TheoBr** [comment]
Just updated docs as well @balazsorban44 , let me know if anything else is needed from me :)

File: src/server/routes/callback.js [hunk 1]
```diff
@@ -396,7 +396,7 @@ export default async function callback(req, res) {
       ...cookies.sessionToken.options,
     })
 
-    await events.signIn({ user, account })
+    await events.signIn({ user, account, profile: null })
```
Comment (balazsorban44) [inline src/server/routes/callback.js:399]: Should be `userObjectReturnedFromAuthorizeHandler` (really sorry for the variable name... 😅) and so please update the documentation and TS types to reflect this. 😊

UPDATE:
Hmm. Revisited the code and actually that variable is used for `user` as well. Should we have the same value twice, or remove profile and mention in the docs that they have user? What does make the most sense in your opinion?

Comment (TheoBr) [reply src/server/routes/callback.js:399]: My mental model associates `profile` with "the thing the auth platform returns", so IMO nulling it out and pointing them to `user` makes more sense.

I'm also _relatively new here_ and _don't use the credentials flow at all_ so I have no strong opinions here

Comment (balazsorban44) [reply src/server/routes/callback.js:399]: Understandable! In that case, I think I would just drop `profile: null` and say in the docs that `user` is what they are looking for if it's a `credentials` provider. If for some reason we add `profile` later on, we could do it in a non-breaking manner then.

Comment (TheoBr) [reply src/server/routes/callback.js]: Just updated @balazsorban44 , ty for the feedback!

**Review by 0ubbe — APPROVED** [review]

**Review by balazsorban44 — APPROVED** [review]
Thank you! 🙂

**Pull request merged**